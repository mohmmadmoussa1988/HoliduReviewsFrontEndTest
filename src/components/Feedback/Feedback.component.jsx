import React from 'react';
import { FeedbackWrapper, Feedback } from './Feedback.styles';
import PositiveFeedback from '../../assets/thumb-up.svg';
import NegativeFeedback from '../../assets/thumb-down.svg';

const Feeback = ({ positiveFeeback, negativeFeedback }) => {
    return (
        <FeedbackWrapper>
            {positiveFeeback &&
                <Feedback>
                    <img src={PositiveFeedback} alt={'positiveFeedback'} />
                    {positiveFeeback}
                </Feedback>
            }
            {negativeFeedback &&
                <Feedback>
                    <img src={NegativeFeedback} alt={'negativeFeedback'} />
                    {negativeFeedback}
                </Feedback>
            }
        </FeedbackWrapper>
    )
}

export default Feeback;