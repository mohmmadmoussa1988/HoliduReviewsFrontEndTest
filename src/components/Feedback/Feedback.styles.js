import styled from "styled-components";
export const FeedbackWrapper = styled.div`
  margin: 19px 0px 0px 0px;
  display: flex;
  flex-direction: column;
  div:first-child {
    margin-bottom: 16px;
  }
`;
export const Feedback = styled.div`
  font-family: "SF-regular";
  font-size: 16px;
  line-height: 24px;
  color: #173753;
  img {
    margin-right: 8px;
    padding-top: 2px;
    vertical-align: top;
  }
`;
