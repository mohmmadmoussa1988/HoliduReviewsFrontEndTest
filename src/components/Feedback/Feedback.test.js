import React from "react";
import { configure, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Feedback from "./Feedback.component";
describe(`<Feedback />`, () => {
  it("should render Feedback", () => {
    const positiveFeeback = "positive feedback";
    const negativeFeedback = "negative feedback";
    const wrapper = shallow(
      <Feedback
        positiveFeeback={positiveFeeback}
        positiveFeeback={negativeFeedback}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
