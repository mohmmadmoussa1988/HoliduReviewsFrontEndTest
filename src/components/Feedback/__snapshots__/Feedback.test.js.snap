// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feedback /> should render Feedback 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="positiveFeedback"
      src="thumb-up.svg"
    />
    negative feedback
  </styled.div>
</styled.div>
`;
